tests/data/hrrr/test_file_loader.py

```diff
@@ -0,0 +1,177 @@
+import unittest
+
+import mock
+import xarray
+
+from tests.RME_test_case import RMETestCase
+from weather_forecast_retrieval.data.hrrr.file_loader import FileLoader
+from weather_forecast_retrieval.data.hrrr.grib_file import GribFile
+from weather_forecast_retrieval.data.hrrr.netcdf_file import NetCdfFile
+
+
+class TestFileLoader(unittest.TestCase):
+    def setUp(self):
+        self.subject = FileLoader()
+
+    def test_file_loader_property(self):
+        self.assertEqual(None, self.subject.file_loader)
+
+        self.subject.file_loader = GribFile
+
+        self.assertEqual(GribFile, self.subject.file_loader)
+
+    def test_file_type(self):
+        self.assertEqual(None, self.subject.file_type)
+
+        self.subject.file_loader = GribFile
+
+        self.assertEqual(GribFile.SUFFIX, self.subject.file_type)
+
+
+def saved_data_return_values():
+    metadata = mock.MagicMock()
+    metadata.name = 'metadata'
+    dataframe = mock.MagicMock(spec={})
+    dataframe.name = 'dataframe'
+    return metadata, dataframe
+
+
+@mock.patch.object(FileLoader, 'get_data')
+@mock.patch.object(
+    FileLoader,
+    'convert_to_dataframes',
+    return_value=saved_data_return_values()
+)
+class TestFileLoaderGetSavedData(RMETestCase):
+    METHOD_ARGS = [
+        RMETestCase.START_DATE, RMETestCase.END_DATE, RMETestCase.BBOX
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        cls.subject = FileLoader()
+
+    def test_parameters(self, _data_patch, _df_patch):
+        self.subject.get_saved_data(*self.METHOD_ARGS, output_dir='path')
+
+        self.assertEqual(self.START_DATE, self.subject.start_date)
+        self.assertEqual(self.END_DATE, self.subject.end_date)
+        self.assertEqual(self.BBOX, self.subject.file_loader.bbox)
+        self.assertEqual('path', self.subject.output_dir)
+
+    def test_defaults_to_grib(self, _data_patch, _df_patch):
+        self.subject.get_saved_data(*self.METHOD_ARGS)
+
+        self.assertIsInstance(self.subject.file_loader, GribFile)
+
+    def test_netcdf_parameter(self, _data_patch, _df_patch):
+        self.subject.get_saved_data(*self.METHOD_ARGS, file_type='netcdf')
+
+        self.assertIsInstance(self.subject.file_loader, NetCdfFile)
+
+    def test_call_get_data(self, data_patch, _df_patch):
```

Comment:
Should this be `_data_patch` to match all the others?

Reply:
This is a flake8 requirement to indicate unused variables. You need to pass them into the test call though, otherwise the mockss will not be effective.